### DISCUSSION TITLE
EmonCMS

### DISCUSSION THREAD

**Discussion by MarcoLeoncino** [arendst/Tasmota#18031] [Q&A]
I am sorry for posting often. Is there the possibility to use EmonCMS with Arendst Tasmota?
Is there a guideline to do that?

**Comment by barbudor** [comment]
AFAIK there is no specific doc covering EmonCMS but I've read this name from time to time
Search the discussions
I may recall that that it is using http but I don't remember if tasmota has to publish using WebQuery or if it's EmonCMS which queries Tasmota.

**Comment by nikito7** [comment]
```
if upsecs%tper==0
and cnt>30
then

ws="WebSend [emoncms.zzzzz.pt:80]"

=>%ws%/input/post.json?json={watts_zero:0,kwh_zero:0,watts_import:%ipwr%,watts_export:%epwr%,kwh_import:%ikw%,kwh_export:%ekw%}&devicekey=%key%&node=%node%

endif
```

**Reply by MarcoLeoncino** [reply]
Thank you!
Shall this script be copied in the console sctipt? What is zzzzz?

**Reply by barbudor** [reply]
It is obviously a sample site. You need to replace with the proper url to your emoncms

This only a piece of a script in Tasmota's scripting language (which is an option not included by default in standard tasmota binaries). It is not a complete solution, a lot is missing to make a full working script. 

Have you searched the discussions? 
For example this one https://github.com/arendst/Tasmota/discussions/11101

**Reply by hallard** [reply]
You can check another full working example here in berry
https://github.com/hallard/WeMos-TIC#send-data-to-emoncms-with-berry-esp32-only

**Comment by MarcoLeoncino** [comment]
Thank you, I will study that... and sorry for my stupid questions...

**Reply by hallard** [reply]
there is no stupid question :-)
